CMSgov/onemac
# add workflow to temp extensions

Story: https://qmacbis.atlassian.net/browse/OY2-16961
Endpoint: https://d3otmxz6odvums.cloudfront.net/

### Details

Add Request Extension to temp extension mini dashboard. After submit of a temp extension, return to the temp extension mini dashboard of the parent (base or renewal waiver).

### Changes

- Added Request Extension button to temp extension mini dashboard
- Updated OneMacForm.tsx to expect location.state.parentId and location.state.parentType
- Updated OneMacForm to replace route landing page placeholders when parentId is set
- More refactoring of submitAny

### Test Plan

1. Login as state submitter
2. Create or navigate to a base waiver details page and the temp extension section on the left nav
3. Verify the button "Request Extension" exists at the top of page 
4. Click the "Request Extension" button
5. Verify you are taken to the temporary extension request form
6. Submit a temporary extension request 
7. Verify after submission you are returned to the parent base waiver temp extension page
8. Verify the success banner appears on the page
9. Verify the newly submitted temp extension request is in the table
